lib/IDE/Formatting.cpp

```diff
@@ -760,6 +769,12 @@ class FormatWalker : public SourceEntityWalker {
   }
 
   bool walkToExprPre(Expr *E) override {
+    if (E->getKind() == ExprKind::StringLiteral &&
+        SM.isBeforeInBuffer(E->getStartLoc(), TargetLocation) &&
+        SM.isBeforeInBuffer(TargetLocation,
+                            Lexer::getLocForEndOfToken(SM, E->getEndLoc()))) {
+      InStringLiteral = true;
+    }
```

Comment:
What about if the line contains a plain string literal say, the line is `"blah"`, will this disable indentation for it ?